include/rocksdb/sst_file_reader.h

```diff
@@ -5,22 +5,60 @@
 #pragma once
 #ifndef ROCKSDB_LITE
 
-#include "rocksdb/sst_dump_tool.h"
-
+#include <inttypes.h>
 #include <memory>
 #include <string>
-#include "db/dbformat.h"
-#include "options/cf_options.h"
-#include "util/file_reader_writer.h"
+#include <vector>
+
+#include "rocksdb/env.h"
+#include "rocksdb/options.h"
+#include "rocksdb/table_properties.h"
+#include "rocksdb/types.h"
 
 namespace rocksdb {
 
+struct ImmutableCFOptions;
+class RandomAccessFileReader;
+class InternalKeyComparator;
+class TableReader;
+struct TableBuilderOptions;
+
+struct DefaultKvHandler {
+  DefaultKvHandler(bool output_hex) : output_hex_(output_hex) {}
+
+  void operator()(const Slice& key, const Slice& value, SequenceNumber sequence,
+                  unsigned char type) {
+    fprintf(stdout, "%s => %s, seq:%" PRIu64 ", type:%d\n",
+            key.ToString(output_hex_).c_str(),
+            value.ToString(output_hex_).c_str(), sequence, type);
+  }
+
+  bool output_hex_;
+};
+
+struct DefaultInfoHandler {
+  void operator()(const std::string& info) {
+    fprintf(stdout, "%s", info.c_str());
+  }
+};
+
+struct DefaultErrHandler {
+  void operator()(const std::string& info) {
+    fprintf(stderr, "%s", info.c_str());
+  }
+};
+
 class SstFileReader {
  public:
-  explicit SstFileReader(const std::string& file_name, bool verify_checksum,
-                         bool output_hex);
-
-  Status ReadSequential(bool print_kv, uint64_t read_num, bool has_from,
+  explicit SstFileReader(
```

Comment:
can we make Options an argument here? It'd also allow us to customize `internal_comparator_` in the constructor, which is needed for a lot of our use cases.